datacube/index/abstract.py

```diff
@@ -1762,15 +1784,16 @@ def search_returning_datasets_light(self,
         """
 
     @abstractmethod
-    def spatial_extent(self, ids: Iterable[DSID], crs: CRS = CRS("EPSG:4326")) -> Optional[Geometry]:
+    def spatial_extent(self, ids: Iterable[DSID], crs: CRS = CRS("EPSG:4326")) -> Geometry | None:
         """
-        Return the combined spatial extent of the nominated datasets.
+        Return the combined spatial extent of the nominated datasets
 
         Uses spatial index.
+
         Returns None if no index for the CRS, or if no identified datasets are indexed in the relevant spatial index.
         Result will not include extents of datasets that cannot be validly projected into the CRS.
 
-        :param ids: An iterable of dataset IDs
+        :param ids: An iterable of dataset IDs (or None)
```

Comment:
`ids` param type doesn't look to be optional

Reply:
Thanks for picking up all these.